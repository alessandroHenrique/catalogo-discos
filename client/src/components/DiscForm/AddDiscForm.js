import React from 'react';
import PropTypes from 'prop-types';

class AddDiscForm extends React.Component {
  state = {
    titleInput: '',
    authorInput: '',
  }

  static propTypes = {
    handleAddDisc: PropTypes.func.isRequired
  }

  handleTitleChange = (e) => {
    const value = e.target.value;
    this.setState(() => ({
      titleInput: value
    }));
  }

  handleAuthorChange = (e) => {
    const value = e.target.value;
    this.setState(() => ({
      authorInput: value
    }));
  }

  handleSubmitForm = (e) => {
    e.preventDefault();
    this.props.handleAddDisc({
      title: this.state.titleInput,
      author: this.state.authorInput,
      collection_id: this.props.collectionId
    });
    this.setState(() => ({
      titleInput: '',
      authorInput: '',
    }));
  }

  render() {
    return (
      <form className="add-option text-left align-items-center add-disc-form justify-content-center" onSubmit={this.handleSubmitForm}>
        <div className="form-group">
          <label htmlFor="title">TÃ­tulo</label>
          <input className="add-option__input" type="text" id="title" onChange={this.handleTitleChange} value={this.state.titleInput} />
        </div>
        <div className="form-group">
          <label htmlFor="author">Autor</label>
          <input className="add-option__input" type="text" id="author" onChange={this.handleAuthorChange} value={this.state.authorInput} />
        </div>

        <button type="submit" className="button">Add Disc</button>
      </form>
    );
  }
}

export default AddDiscForm;
