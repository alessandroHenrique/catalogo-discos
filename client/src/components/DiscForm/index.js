import { connect } from 'react-redux';
import { addDisc } from '../../actions/discs';
import AddDiscForm from './AddDiscForm';

const mapDispatchToProps = (dispatch) => ({
  handleAddDisc: (disc) => dispatch(addDisc(disc))
});

export default connect(null, mapDispatchToProps)(AddDiscForm);
