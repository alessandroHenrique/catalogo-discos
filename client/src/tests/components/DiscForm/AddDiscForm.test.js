import DiscForm from '../../../components/DiscForm/AddDiscForm';
import React from 'react';
import renderer from 'react-test-renderer';


describe('DiscForm', () => {
  it('should render add disc form', () => {

    const component = renderer.create(
      <DiscForm
        handleAddCollection={() => {}}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
