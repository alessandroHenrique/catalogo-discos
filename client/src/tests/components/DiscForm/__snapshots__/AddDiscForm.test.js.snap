// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscForm should render add disc form 1`] = `
<form
  className="add-option text-left align-items-center add-disc-form justify-content-center"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      TÃ­tulo
    </label>
    <input
      className="add-option__input"
      id="title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="author"
    >
      Autor
    </label>
    <input
      className="add-option__input"
      id="author"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    className="button"
    type="submit"
  >
    Add Disc
  </button>
</form>
`;
